### PR TITLE
add torch_clamp

### LINKED ISSUE #7751: clamp

### PR DESCRIPTION
**Pull request by SaabriinMo**
Close #7751

### CODE REVIEW

**Comment by SaabriinMo** [comment]
Hello,
The github test are failing for some weird reason but when i run it in my local machine i get this response:

============================= test session starts ==============================
platform linux -- Python 3.8.10, pytest-7.1.2, pluggy-1.0.0 -- /usr/bin/python3
cachedir: .pytest_cache
metadata: {'Python': '3.8.10', 'Platform': 'Linux-5.15.49-linuxkit-x86_64-with-glibc2.29', 'Packages': {'pytest': '7.1.2', 'pluggy': '1.0.0'}, 'Plugins': {'metadata': '2.0.4', 'typeguard': '2.13.3', 'hypothesis': '6.55.0', 'json-report': '1.5.0'}}
hypothesis profile 'ivy_profile' -> max_examples=5, print_blob=True, deadline=timedelta(milliseconds=500000), suppress_health_check=[HealthCheck.too_slow, HealthCheck.filter_too_much], database=MultiplexedDatabase(DirectoryBasedExampleDatabase('/ivy/.hypothesis/examples/'), ReadOnlyDatabase(RedisExampleDatabase(Redis<ConnectionPool<Connection<host=redis-17011.c259.us-central1-2.gce.cloud.redislabs.com,port=17011,db=0>>>, expire_after=datetime.timedelta(days=8))))
rootdir: /ivy
plugins: metadata-2.0.4, typeguard-2.13.3, hypothesis-6.55.0, json-report-1.5.0
collecting ... collected 1 item

ivy_tests/test_ivy/test_frontends/test_torch/test_tensor.py::test_torch_instance_clamp[cpu-ivy.functional.backends.torch-False-False] 

=============================== warnings summary ===============================
../usr/local/lib/python3.8/dist-packages/flatbuffers/compat.py:19
  /usr/local/lib/python3.8/dist-packages/flatbuffers/compat.py:19: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
    import imp

ivy_tests/test_ivy/test_frontends/test_torch/test_tensor.py::test_torch_instance_clamp[cpu-ivy.functional.backends.torch-False-False]
ivy_tests/test_ivy/test_frontends/test_torch/test_tensor.py::test_torch_instance_clamp[cpu-ivy.functional.backends.torch-False-False]
ivy_tests/test_ivy/test_frontends/test_torch/test_tensor.py::test_torch_instance_clamp[cpu-ivy.functional.backends.torch-False-False]
ivy_tests/test_ivy/test_frontends/test_torch/test_tensor.py::test_torch_instance_clamp[cpu-ivy.functional.backends.torch-False-False]
ivy_tests/test_ivy/test_frontends/test_torch/test_tensor.py::test_torch_instance_clamp[cpu-ivy.functional.backends.torch-False-False]
  /ivy/ivy_tests/test_ivy/helpers/function_testing.py:1342: UserWarning: To copy construct from a tensor, it is recommended to use sourceTensor.clone().detach() or sourceTensor.clone().detach().requires_grad_(True), rather than torch.tensor(sourceTensor).
    ins_gt = frontend_creation_fn(

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
======================== 1 passed, 6 warnings in 10.55s ========================

File: ivy_tests/test_ivy/test_frontends/test_torch/test_tensor.py [hunk 1]
```diff
@@ -3608,6 +3606,67 @@ def test_torch_instance_cumsum(
...
 
 
+@st.composite
+def _get_clamp_inputs(draw):
+    shape = draw(
+        helpers.get_shape()
+    )
+    x_dtype, x = draw(
+        helpers.dtype_and_values(
+            available_dtypes=helpers.get_dtypes("float"),
```
Comment (CatB1t) [inline ivy_tests/test_ivy/test_frontends/test_torch/test_tensor.py:3616]: We should test for integers too, `torch.clamp` accepts both `float` and `int` inputs.

File: ivy_tests/test_ivy/test_frontends/test_torch/test_tensor.py [hunk 2]
```diff
@@ -3608,6 +3606,67 @@ def test_torch_instance_cumsum(
...
+            min_value=-1e04,
```
Comment (CatB1t) [inline ivy_tests/test_ivy/test_frontends/test_torch/test_tensor.py:3617]: We shouldn't limit the generated values.

File: ivy_tests/test_ivy/test_frontends/test_torch/test_tensor.py [hunk 3]
```diff
@@ -3608,6 +3606,67 @@ def test_torch_instance_cumsum(
...
+    )
+    return x_dtype, x, min, max
+
+
+# clamp
+@handle_frontend_method(
+    class_tree=CLASS_TREE,
+    init_tree="torch.tensor",
+    method_name="clamp",
+    dtype_and_x_min_max=_get_clamp_inputs().filter(lambda x: "float16" not in x[0])
```
Comment (CatB1t) [inline ivy_tests/test_ivy/test_frontends/test_torch/test_tensor.py:3637]: We shouldn't filter the data types generated from the helper, they won't be generated if the method does not support it, we should flag it using the `unsupported` decorator.

**Review by CatB1t — CHANGES_REQUESTED** [review]
Hello @SaabriinMo, thanks for your contribution! can you please revert the changes to the formatting that aren't related to?
